extensions/guacamole-vault/modules/guacamole-vault-ksm/src/main/java/org/apache/guacamole/vault/ksm/secret/KsmClient.java

```diff
@@ -399,32 +497,75 @@ public KeeperRecord getRecordByHost(String hostname) throws GuacamoleException {
     }
 
     /**
-     * Returns the record associated with the given username. If no such record
-     * exists, or there are multiple such records, null is returned.
+     * Returns the record associated with the given username and domain. If no
+     * such record exists, or there are multiple such records, null is returned.
      *
      * @param username
      *     The username of the record to return.
      *
+     * @param domain
+     *     The domain of the record to return.
```

Comment:
Hmm, what specifically would you request that I add? All it means if this parameter is null is that it will fetch records for which the domain is null. It's just regular equality, nothing special.

Reply:
I think just adding `..., which may be null.` or `..., or null if there is no domain.` would be sufficient. Otherwise, it sounds like a domain is always required, and it's unclear that `null` is an expected value if there is no domain.

Reply:
Fair - added.